tests/scripts/all.sh

```diff
@@ -1221,11 +1221,12 @@ component_test_cmake_out_of_source () {
     # file is missing (ssl-opt.sh tolerates the absence of some files so
     # may exit with status 0 but emit errors).
     if_build_succeeded ./tests/ssl-opt.sh -f 'Fallback SCSV: beginning of list' 2>ssl-opt.err
+    # Record an error if ssl-opt.err is non-empty. Print it out first.
     if [ -s ssl-opt.err ]; then
         cat ssl-opt.err >&2
-        record_status [ ! -s ssl-opt.err ]
-        rm ssl-opt.err
+        record_status [ -s ssl-opt.err ]
```

Comment:
This will always succeed, which I think is the opposite of what we want. Either way, I find this use of `record_status` a bit hackish, so either we can do something cleaner, or we should at least add a comment to document the hack, eg "we know this will always fail, we just want the name of the command in the error message".

Reply:
Uh, woops, yes, I broke that. And yes, it's a hack, and in this PR I added a comment to explain it, but I broke it!

Reply:
I rewrote the history to amend the commit that removes the `!`. As a bonus the hack is now less hackish.

Reply:
Thanks. Before seeing your fix I didn't realise this hack was useless, as `cat empty-file` will do nothing (visible).